cpp/src/strings/search/findall.cu

```diff
@@ -126,6 +126,43 @@ std::unique_ptr<column> findall(strings_column_view const& input,
                            mr);
 }
 
+namespace {
+struct find_re_fn {
+  column_device_view d_strings;
+
+  __device__ size_type operator()(size_type const idx,
+                                  reprog_device const prog,
+                                  int32_t const thread_idx) const
```

Comment:
Why is this `int32_t` and not `cudf::thread_index_type`?

Reply:
Yes, this is used for managing working memory and so has no correspondence with either `size_type` or `thread_index_type`. The function signature is required for the regex utilities and outside the scope of this PR.